src/Viewer/services/decoder/decodeWorker.js

```diff
@@ -0,0 +1,55 @@
+import Database from "../Database";
+import {DataInputStream, DataInputStreamEOFError} from "./DataInputStream";
+import FourByteClpIrStreamReader from "./FourByteClpIrStreamReader";
+import ResizableUint8Array from "./ResizableUint8Array";
+
+const decodePage = async (fileName, logEvents, inputStream, page) => {
+    const dataInputStream = new DataInputStream(inputStream);
+    const _outputResizableBuffer = new ResizableUint8Array(inputStream.byteLength);
+    const _irStreamReader = new FourByteClpIrStreamReader(dataInputStream, null);
+
+    const _logEventMetadata = [];
+    for (let i = 0; i < logEvents.length; i++) {
+        const decoder = _irStreamReader._streamProtocolDecoder;
+        decoder._setTimestamp(logEvents[i].prevTs);
+
+        try {
+            _irStreamReader.readAndDecodeLogEvent(
+                _outputResizableBuffer,
+                _logEventMetadata
+            );
+        } catch (error) {
+            // Ignore EOF errors since we should still be able
+            // to print the decoded messages
+            if (error instanceof DataInputStreamEOFError) {
+                // TODO Give visual indication that the stream is truncated
+                console.error("Stream truncated.");
+            } else {
+                console.log("random error");
+                throw error;
```

Comment:
We should do some more meaningful logging here...

Reply:
Should we add more cases for error catching? If so, what kind of error are we expecting?